vmupdate/vmupdate.py

```diff
@@ -114,17 +115,17 @@ def get_targets(args, app):
                 f": {', '.join(unknowns) if plural else ''.join(unknowns)}"
             )
     else:
-        targets += smart_targeting(app, args)
+        targets.update(smart_targeting(app, args))
 
     # remove skipped qubes and dom0 - not a target
     to_skip = args.skip.split(',')
-    targets = [vm for vm in targets
-               if vm.name != 'dom0' and vm.name not in to_skip]
+    targets = set([vm for vm in targets
```

Comment:
In fact, maybe earlier use of `set()` is not necessary at all, since you make it all set here?